### PR TITLE
cmd/containerboot: add tests.

### PR DESCRIPTION
**Pull request by danderson**
Right now just one test, because it's already a lot of scaffolding code to look at. In a followup I will table-ize the test and exercise other configurations.

### CODE REVIEW

File: cmd/containerboot/main_test.go [hunk 1]
```diff
@@ -0,0 +1,438 @@
...
+func (k *kubeServer) Reset() {
+	k.Lock()
+	defer k.Unlock()
+	k.secret = map[string]string{}
+}
+
+func (k *kubeServer) Start() error {
+	root := filepath.Join(k.FSRoot, "var/run/secrets/kubernetes.io/serviceaccount")
+
+	cert, err := mkSelfSigned(filepath.Join(root, "ca.crt"))
```
Comment (maisem) [inline cmd/containerboot/main_test.go:283]: `httptest.Server.StartTLS` _might_ be better.

File: cmd/containerboot/main_test.go [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+			if strings.HasSuffix(line, " "+want) {
+				return
+			}
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+	t.Fatalf("timed out waiting for wanted output line %q. Output:\n%s", want, b.String())
+}
+
+func waitArgs(t *testing.T, timeout time.Duration, root, path, wantArgs string) {
```
Comment (maisem) [inline cmd/containerboot/main_test.go:146]: all args except root are self-explanatory.

Comment (maisem) [reply cmd/containerboot/main_test.go:146]: add a comment?

**Review by maisem — APPROVED** [review]

File: cmd/containerboot/main_test.go (hunk 1 above)
Comment (danderson) [reply cmd/containerboot/main_test.go:283]: Replaced with httptest.Server.

File: cmd/containerboot/main_test.go (hunk 2 above)
Comment (danderson) [reply cmd/containerboot/main_test.go:146]: Documented both wait funcs, and renamed `root` to `removeStr`.

**Pull request merged**